app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -794,39 +476,589 @@ public void onCharacteristicChanged(BluetoothGatt gatt, BluetoothGattCharacteris
             final PowerManager.WakeLock wakeLock1 = JoH.getWakeLock("DexCollectionService", 60000);
             try {
                 final byte[] data = characteristic.getValue();
-                final String hexdump = HexDump.dumpHexString(data);
-                if (!hexdump.contains("0x00000000 00      ")) {
-                    static_last_hexdump = hexdump;
-                }
-                Log.i(TAG, "onCharacteristicChanged entered " + hexdump);
+
                 if (data != null && data.length > 0) {
                     setSerialDataToTransmitterRawData(data, data.length);
+
+                    final String hexdump = HexDump.dumpHexString(data);
+                    //if (!hexdump.contains("0x00000000 00      ")) {
+                    if (data.length > 1 || data[0] != 0x00) {
+                        static_last_hexdump = hexdump;
+                    }
+                    if (d) Log.i(TAG, "onCharacteristicChanged entered " + hexdump);
+
                 }
                 lastdata = data;
-                setFailoverTimer(); // restart the countdown
-                // intentionally left hanging wakelock for 5 seconds after we receive something
-                final PowerManager.WakeLock wakeLock2 = JoH.getWakeLock("DexCollectionLinger", 5000);
+
```

Comment:
Here the creation of the `FailoverTimer()` is set by `Inevitable.task()` so only on the last received bluetooth packet where there is then 1 second of following silence.